<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Empirical testing of DNA match probabilities • DNAtools</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Empirical testing of DNA match probabilities">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DNAtools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/db_vignette.html">Empirical testing of DNA match probabilities</a>
    </li>
    <li>
      <a href="../articles/dna_vignette.html">STR markers in forensic genetics</a>
    </li>
    <li>
      <a href="../articles/noa_vignette.html">On the exact distribution of the numbers of alleles in DNA mixtures</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Empirical testing of DNA match probabilities</h1>
                        <h4 class="author">Mikkel Meyer Andersen, James Michael Curran and Torben Tvedebrink</h4>
            
            <h4 class="date">2019-12-19</h4>
      
      
      <div class="hidden name"><code>db_vignette.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>In 2001, a poster was presented by a forensic scientist from Arizona <span class="citation">(Troyer, Gilboy, and Koeneman 2001)</span> at a scientific meeting on human identification. This poster reported a nine locus match between two unrelated men, one white and one black <span class="citation">(Kaye 2009)</span>. It was not a full match. Both men had been typed at thirteen loci in total, and <em>partially matched</em> at three of the remaining four loci. These partially or non-matching loci would have excluded either man as a suspect if the other was the true offender. However, such a match seemed to be at odds with the random match probabilities. On one hand, these two men were in a DNA database which consisted of approximately 65,000 profiles, and on the other hand, the random match probabilities for the nine locus genotype were <em>1 in 754 million in Caucasians, 1 in 561 billion in African Americans, and 1 in 113 trillion in Southwest Hispanics</em>, <span class="citation">(Troyer, Gilboy, and Koeneman 2001)</span>.</p>
<p>As we will show later this is, in effect, an example of the <em>birthday problem</em> and therefore is regarded as completely predictable from a statistical perspective. However, most us who have taught a class on the birthday problem know that our students are initially sceptical.</p>
<div id="dna-evidence-matches-and-partial-matches" class="section level2">
<h2 class="hasAnchor">
<a href="#dna-evidence-matches-and-partial-matches" class="anchor"></a>DNA evidence, matches and partial matches</h2>
<p>A pair of profiles is said to (fully) match if every allele at every locus that occurs in one profile occurs in the other. A pair of profiles are said to partially match if there are allelic matches at a subset of loci. <span class="citation">Weir (2004)</span> provided a taxonomy for describing partial matches which depends on the number of fully, partial and non-matching loci between a pair of profiles. For any given pair of (full) profiles from the same multiplex there will be: <span class="math inline">\(m\)</span> loci where both alleles match, <span class="math inline">\(p\)</span> loci where only one of the alleles matches, and <span class="math inline">\(L-m-p\)</span> loci where none of the alleles match, where <span class="math inline">\(L\)</span> is the total number of loci. For example, the profile that <span class="citation">Troyer, Gilboy, and Koeneman (2001)</span> found was a <em>9/3</em> partial match on 13 loci - nine fully matching loci (<span class="math inline">\(m=9\)</span>), three partially matching loci (<span class="math inline">\(p=3\)</span>), and one non-matching locus.</p>
</div>
<div id="dna-database-comparison-exercises" class="section level2">
<h2 class="hasAnchor">
<a href="#dna-database-comparison-exercises" class="anchor"></a>DNA database comparison exercises</h2>
<p>The Troyer-match came from a database matching exercise. In such an exercise every profile is compared with every other profile in the database. This type of comparison exercise is absolutely essential and, in addition, can provide some interesting information about the statistical properties of the population under consideration. We say that database comparison is essential in the first instance for the detection of duplicates. Duplicates may arise in a number of different ways. For example, an offender may provide a false name or an offender’s name may be entered incorrectly. Alternatively, an offender may have an identical twin who is already in the DNA database. There are six pairs of identical twins in the New Zealand National DNA Database (NZDNADB). Forensic scientists are also interested in ‘very close’ matches. For example, a pair of profiles might fully match at nine loci out of ten and partially match at the remaining locus. This may happen either because the donors of the samples are very close relatives. It is more likely, however, that the profiles do not match because of allelic dropout, primer binding site mutations, nomenclature changes or somatic mutation.</p>
</div>
<div id="the-birthday-problem" class="section level2">
<h2 class="hasAnchor">
<a href="#the-birthday-problem" class="anchor"></a>The birthday problem</h2>
<p><span class="citation">Weir (2007)</span> and others <span class="citation">(Brenner 2007; Curran, Walsh, and Buckleton 2007; Mueller 2008; Kaye 2009)</span> note that the presence of matching profiles in a DNA database is effectively an instance of the well-known <em>birthday problem</em> <span class="citation">(2010)</span> where, in a group of at least 23 randomly chosen people, there is a greater than 50% chance that one pair of them will have the same birthday. Early critics, implicitly calculating the expected number of matches as <span class="math inline">\(N\pi\)</span>, used the wrong value for <span class="math inline">\(N\)</span> and the wrong value for <span class="math inline">\(\pi\)</span>. Firstly, the number of pairwise matches, not the size of the database, is the relevant quantity. Although the database size is relatively small, the number of pairwise comparisons is very large. The Arizona database contained of <span class="math inline">\(N=65,493\)</span> profiles <span class="citation">(Brenner 2007)</span>. Therefore, there are <span class="math display">\[
N_\text{Comparisons}=\frac{N(N-1)}{2} = 2,144,633,778
\]</span> or approximately two billion, possible pairwise comparisons.</p>
<div id="the-pairwise-profile-matching-probability" class="section level3">
<h3 class="hasAnchor">
<a href="#the-pairwise-profile-matching-probability" class="anchor"></a>The pairwise profile matching probability</h3>
<p>Secondly, the random match probability is not the probability we need. The random match probability for the pair of profiles in question answers the question <em>“What is the probability that someone other than these two men would have this particular nine locus profile.”</em> The probability we actually want is <em>“What is the probability that two randomly selected profiles would match at nine loci, partially match at three loci, and not match at one locus.”</em> . At a given locus the two profiles may either match (<span class="math inline">\(m/p=1/0\)</span>), partially match (<span class="math inline">\(m/p=0/1\)</span>) or mismatch (<span class="math inline">\(m/p=0/0\)</span>). <span class="citation">Weir (2004)</span>, <span class="citation">Weir (2007)</span> showed that the probabilities of mismatch (<span class="math inline">\(P_{0/0}\)</span>), partial match (<span class="math inline">\(P_{0/1}\)</span>) and match (<span class="math inline">\(P_{1/0}\)</span>) could be expressed by <span class="math display">\[\begin{align*}
   P(\text{Mismatch}) &amp;= P_{0/0} =  
  D^{-1}\{\theta^2(1{-}\theta)(1{-}S_2) +
  2\theta(1{-}\theta)^2(1{-}2S_2{+}S_3) + (1{-}\theta)^3(1{-}4S_2{+}4S_3{+}2S_2^2{-}3S_4)\},\\
   P(\text{Partial match}) &amp;=  P_{0/1} = 
   D^{-1}\{8\theta^2(1{-}\theta)(1{-}S_2) + 4\theta(1{-}\theta)^2(1{-}S_3)
    + 4(1{-}\theta)^3(S_2{-}S_3{-}S_2^2{+}S_4)\},\\ 
  P(\text{Match}) &amp;=  P_{1/0} = 
    D^{-1}\{6\theta^3 + \theta^2(1{-}\theta)(2{+}9S_2) +
    2\theta(1{-}\theta)^2(2S_2{+}S_3) +
    (1{-}\theta)^3(2S_2^2{-}S_4)\},
\end{align*}\]</span> where <span class="math inline">\(D=(1{+}2\theta)(1{+}\theta)\)</span> and <span class="math inline">\(S_k=\sum_{i}p_i^k\)</span>, with <span class="math inline">\(p_i\)</span> being the frequency of the <span class="math inline">\(i\)</span> allele.</p>
<p><span class="citation">Tvedebrink (2010)</span> showed that the overall probability of matching at <span class="math inline">\(m\)</span> and partially matching at <span class="math inline">\(p\)</span> loci, <span class="math inline">\(\pi_{m/p}\)</span>, can be calculated by recursion over loci:</p>
<p><span class="math display">\[
  \pi^{\ell+1}_{m/p} = \left\{\begin{array}{ll}
      P^{\ell+1}_{0/0}\pi^\ell_{m/p} +
      P^{\ell+1}_{0/1}\pi^\ell_{m/p-1} +
      P^{\ell+1}_{1/0}\pi^\ell_{m-1/p}&amp; m&gt;0~\text{and}~p&gt;0,\\
      P^{\ell+1}_{0/0}\pi^\ell_{0/p} + P^{\ell+1}_{0/1}\pi^\ell_{0/p{-}1}&amp;m=0~\text{and}~p&gt;0,\\
      P^{\ell+1}_{0/0}\pi^\ell_{m/0} +
      P^{\ell+1}_{1/0}\pi^\ell_{m{-}1/0}&amp;m&gt;0~\text{and}~p=0\\
      P^{\ell+1}_{0/0}\pi^\ell_{0/0}&amp; m=0~\text{and}~p=0,  
      \end{array}\right.
\]</span> where the <em>sum</em> of the subscripts for each term on the right hand side equals the subscript on the left hand side, e.g. the subscripts of the last term in the first equation gives <span class="math inline">\(1/0 + m{-}1/p = m/p\)</span>. The initial step of the recursion has <span class="math inline">\(\pi^1_{1/0} = P^1_{1/0}\)</span>, <span class="math inline">\(\pi^1_{0/1} = P^1_{0/1}\)</span> and <span class="math inline">\(\pi^1_{0/0} = P^1_{0/0}\)</span>.</p>
<p>The coancestry coefficient, <span class="math inline">\(\theta\)</span> or <span class="math inline">\(F_{ST}\)</span> models low levels of relatedness between individuals in the same subpopulation, and is typically between 0 and 0.03.</p>
</div>
</div>
<div id="using-the-package-dnatools" class="section level2">
<h2 class="hasAnchor">
<a href="#using-the-package-dnatools" class="anchor"></a>Using the package <code>DNAtools</code>
</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(DNAtools)</a></code></pre></div>
<p>The expected data format of the databases used as input for the functions in <code>DNAtools</code> is a data frame, which is constituted by a column of DNA profile identifiers (the first column) and two columns per typed DNA marker. An example is given below:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(dbExample, <span class="dt">package =</span> <span class="st">"DNAtools"</span>)</a>
<a class="sourceLine" id="cb2-2" title="2">knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(dbExample)[,<span class="dv">1</span><span class="op">:</span><span class="dv">9</span>])</a></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">id</th>
<th align="left">D16S539.1</th>
<th align="left">D16S539.2</th>
<th align="left">D18S51.1</th>
<th align="left">D18S51.2</th>
<th align="left">D19S433.1</th>
<th align="left">D19S433.2</th>
<th align="left">D21S11.1</th>
<th align="left">D21S11.2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">11</td>
<td align="left">11</td>
<td align="left">15</td>
<td align="left">21</td>
<td align="left">14</td>
<td align="left">14</td>
<td align="left">28</td>
<td align="left">29</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">13</td>
<td align="left">12</td>
<td align="left">15</td>
<td align="left">14</td>
<td align="left">16</td>
<td align="left">16</td>
<td align="left">29</td>
<td align="left">28</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="left">9</td>
<td align="left">9</td>
<td align="left">13</td>
<td align="left">17</td>
<td align="left">14</td>
<td align="left">14</td>
<td align="left">28</td>
<td align="left">27</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="left">11</td>
<td align="left">12</td>
<td align="left">14</td>
<td align="left">15</td>
<td align="left">15</td>
<td align="left">13</td>
<td align="left">32</td>
<td align="left">29</td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="left">12</td>
<td align="left">12</td>
<td align="left">17</td>
<td align="left">12</td>
<td align="left">15.2</td>
<td align="left">13</td>
<td align="left">31.2</td>
<td align="left">28</td>
</tr>
<tr class="even">
<td align="left">6</td>
<td align="left">9</td>
<td align="left">13</td>
<td align="left">17</td>
<td align="left">14</td>
<td align="left">13</td>
<td align="left">14</td>
<td align="left">30.2</td>
<td align="left">28</td>
</tr>
</tbody>
</table>
<p>From which we estimate allele frequencies for later use:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1">allele_freqs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">lapply</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="cf">function</span>(x){</a>
<a class="sourceLine" id="cb3-2" title="2">  al_freq &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(dbExample[[x<span class="op">*</span><span class="dv">2</span>]], dbExample[[<span class="dv">1</span><span class="op">+</span>x<span class="op">*</span><span class="dv">2</span>]]))<span class="op">/</span>(<span class="dv">2</span><span class="op">*</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(dbExample))</a>
<a class="sourceLine" id="cb3-3" title="3">  al_freq[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/order">sort.list</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(al_freq)))]</a>
<a class="sourceLine" id="cb3-4" title="4">})</a>
<a class="sourceLine" id="cb3-5" title="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(allele_freqs) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">sub</a></span>(<span class="st">"</span><span class="ch">\\</span><span class="st">.1"</span>, <span class="st">""</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(dbExample)[(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>)<span class="op">*</span><span class="dv">2</span>])</a></code></pre></div>
<p>We use the example database provided by <code><a href="https://www.rdocumentation.org/packages/utils/topics/data">data(dbExample, package = "DNAtools")</a></code> to show the usage of the functions of <code>DNAtools</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1">db_summary &lt;-<span class="st"> </span><span class="kw"><a href="../reference/dbCompare.html">dbCompare</a></span>(dbExample, <span class="dt">hit =</span> <span class="dv">6</span>, <span class="dt">trace =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="right">0</th>
<th align="right">1</th>
<th align="right">2</th>
<th align="right">3</th>
<th align="right">4</th>
<th align="right">5</th>
<th align="right">6</th>
<th align="right">7</th>
<th align="right">8</th>
<th align="right">9</th>
<th align="right">10</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>0</strong></td>
<td align="right">102</td>
<td align="right">1368</td>
<td align="right">7122</td>
<td align="right">21878</td>
<td align="right">44189</td>
<td align="right">59463</td>
<td align="right">54601</td>
<td align="right">34203</td>
<td align="right">13571</td>
<td align="right">3281</td>
<td align="right">353</td>
</tr>
<tr class="even">
<td><strong>1</strong></td>
<td align="right">206</td>
<td align="right">2114</td>
<td align="right">10013</td>
<td align="right">26084</td>
<td align="right">43656</td>
<td align="right">47418</td>
<td align="right">34320</td>
<td align="right">15463</td>
<td align="right">4145</td>
<td align="right">472</td>
<td align="right"></td>
</tr>
<tr class="odd">
<td><strong>2</strong></td>
<td align="right">165</td>
<td align="right">1477</td>
<td align="right">5710</td>
<td align="right">12566</td>
<td align="right">17049</td>
<td align="right">14642</td>
<td align="right">7570</td>
<td align="right">2220</td>
<td align="right">310</td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="even">
<td><strong>3</strong></td>
<td align="right">72</td>
<td align="right">556</td>
<td align="right">1821</td>
<td align="right">3250</td>
<td align="right">3361</td>
<td align="right">2135</td>
<td align="right">719</td>
<td align="right">116</td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="odd">
<td><strong>4</strong></td>
<td align="right">22</td>
<td align="right">149</td>
<td align="right">360</td>
<td align="right">493</td>
<td align="right">379</td>
<td align="right">156</td>
<td align="right">34</td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="even">
<td><strong>5</strong></td>
<td align="right">6</td>
<td align="right">19</td>
<td align="right">44</td>
<td align="right">41</td>
<td align="right">26</td>
<td align="right">5</td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="odd">
<td><strong>6</strong></td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="even">
<td><strong>7</strong></td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="odd">
<td><strong>8</strong></td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="even">
<td><strong>9</strong></td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="odd">
<td><strong>10</strong></td>
<td align="right">0</td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
</tr>
</tbody>
</table>
<p>The <code>hit</code> argument returns pairs of profiles that fully match at <code>hit</code> (here 6) or more loci.</p>
<p>There is a  method for the returned object. Applying this method to  yields the <em>dropping ball</em>-picture below. The right end of the <em>distribution</em> is interesting part, due to the larger number of coinciding loci between profile pairs.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(db_summary)</a></code></pre></div>
<p><img src="db_vignette_files/figure-html/unnamed-chunk-6-1.png" width="768"></p>
<p>By using the <code>dbExpect</code> and <code>dbVariance</code> functions we can compute the expected counts and standard deviations for the number of matching profiles at various levels. Both functions implements the recursions formulae of <span class="citation">Tvedebrink et al. (2012)</span>, which enables fast and accurate evaluations of the expectations and variances. The <code>dbExpect</code>-function takes the locus-wise allele frequencies and number of profiles as arguments. Furthermore, the <span class="math inline">\(\theta\)</span>-correction is adjusted for if <code>theta &gt; 0</code> and adjustment for close relatedness can be done by the <code>k</code>-argument, which is a three-dimensional vector with non-negative entries summing to one (cf. below).</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1">db_expect &lt;-<span class="st"> </span><span class="kw"><a href="../reference/dbExpect.html">dbExpect</a></span>(allele_freqs, <span class="dt">n =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(dbExample), <span class="dt">theta =</span> <span class="dv">0</span>, <span class="dt">vector =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb6-2" title="2">db_sd &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">sqrt</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/diag">diag</a></span>(<span class="kw"><a href="../reference/dbVariance.html">dbVariance</a></span>(allele_freqs, <span class="dt">n =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(dbExample), <span class="dt">theta =</span> <span class="dv">0</span>)))</a>
<a class="sourceLine" id="cb6-3" title="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(db_summary)</a>
<a class="sourceLine" id="cb6-4" title="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(db_expect, <span class="dt">pch =</span> <span class="dv">4</span>)</a>
<a class="sourceLine" id="cb6-5" title="5"><span class="cf">for</span>(x <span class="cf">in</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/seq">seq_along</a></span>(db_sd)) <span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/segments">segments</a></span>(<span class="dt">x0 =</span> x, <span class="dt">y0 =</span> db_expect[x]<span class="op">-</span><span class="dv">2</span><span class="op">*</span>db_sd[x], <span class="dt">y1 =</span> db_expect[x]<span class="op">+</span><span class="dv">2</span><span class="op">*</span>db_sd[x])</a>
<a class="sourceLine" id="cb6-6" title="6"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/legend">legend</a></span>(<span class="st">"topright"</span>, <span class="dt">bty =</span> <span class="st">"n"</span>, <span class="dt">pch =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>,<span class="dv">4</span>,<span class="ot">NA</span>), <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Observed"</span>, <span class="st">"Expected"</span>, <span class="st">"95%-CI"</span>),</a>
<a class="sourceLine" id="cb6-7" title="7">       <span class="dt">lty =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="ot">NA</span>,<span class="ot">NA</span>,<span class="dv">1</span>))</a></code></pre></div>
<p><img src="db_vignette_files/figure-html/unnamed-chunk-7-1.png" width="768"></p>
<p>Superimposed are the expected counts and associated 95% approximate confidence intervals. The labels of the first axis denote the number of matching and partial-matching loci. The second axis is on a <span class="math inline">\(\log_{10}\)</span>-scale.</p>
<p>Often it is of interest to look at the number of matching alleles rather than on the number of matching and partial matching loci. As the number of matching alleles is given by <span class="math inline">\(a = 2m+p\)</span> it is straight forward to look at these instead. The function <code>dbCollapse</code> sums over combinations of <span class="math inline">\(m\)</span> and <span class="math inline">\(p\)</span> resulting in <span class="math inline">\(a\)</span> alleles, for <span class="math inline">\(a = 0, \dots, 2L\)</span>:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1">a_expect &lt;-<span class="st"> </span><span class="kw"><a href="../reference/dbCollapse.html">dbCollapse</a></span>(<span class="kw"><a href="../reference/dbExpect.html">dbExpect</a></span>(<span class="dt">probs =</span> allele_freqs,  <span class="dt">n =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(dbExample)))</a>
<a class="sourceLine" id="cb7-2" title="2">a_observed &lt;-<span class="st"> </span><span class="kw"><a href="../reference/dbCollapse.html">dbCollapse</a></span>(<span class="kw"><a href="../reference/dbCompare.html">dbCompare</a></span>(dbExample, <span class="dt">trace =</span> <span class="ot">FALSE</span>)<span class="op">$</span>m)</a>
<a class="sourceLine" id="cb7-3" title="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/seq">seq_along</a></span>(a_observed)<span class="op">-</span><span class="dv">1</span>, a_observed, <span class="dt">log =</span> <span class="st">"y"</span>,</a>
<a class="sourceLine" id="cb7-4" title="4">     <span class="dt">xlab =</span> <span class="st">"Number of matching alleles"</span>, <span class="dt">ylab =</span> <span class="st">"Count"</span>)</a>
<a class="sourceLine" id="cb7-5" title="5">Warning <span class="cf">in</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/grDevices/topics/xy.coords">xy.coords</a></span>(x, y, xlabel, ylabel, log)<span class="op">:</span><span class="st"> </span><span class="dv">5</span> y values <span class="op">&lt;=</span><span class="st"> </span><span class="dv">0</span> omitted from</a>
<a class="sourceLine" id="cb7-6" title="6">logarithmic plot</a>
<a class="sourceLine" id="cb7-7" title="7"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/seq">seq_along</a></span>(a_expect)<span class="op">-</span><span class="dv">1</span>, a_expect, <span class="dt">pch =</span> <span class="dv">4</span>)</a></code></pre></div>
<p><img src="db_vignette_files/figure-html/unnamed-chunk-8-1.png" width="768"></p>
<p>The table below lists the <strong>k</strong>-vector for some typical types of relationsships, where k<sub>i</sub> is the probability of having <em>i</em> alleles Identical-by-descent (IBD).</p>
<table class="table">
<colgroup>
<col width="18%">
<col width="18%">
<col width="18%">
<col width="17%">
<col width="14%">
<col width="14%">
</colgroup>
<thead><tr class="header">
<th align="left">Relationship</th>
<th align="left">Full-siblings (FS)</th>
<th align="left">First-cousins (FC)</th>
<th align="left">Parent-child (PC)</th>
<th align="left">Avuncular (AV)</th>
<th align="left">Unrelated (UN)</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">
<strong>k</strong>=(k<sub>2</sub>,k<sub>1</sub>,k<sub>0</sub>)</td>
<td align="left">(0.25, 0.5, 0.25)</td>
<td align="left">(0, 0.25, 0.75)</td>
<td align="left">(0, 1, 0)</td>
<td align="left">(0, 0.5, 0.5)</td>
<td align="left">(0, 0, 1)</td>
</tr></tbody>
</table>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1">relatives &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(</a>
<a class="sourceLine" id="cb8-2" title="2">  <span class="dt">UN =</span> <span class="kw"><a href="../reference/dbExpect.html">dbExpect</a></span>(<span class="dt">probs =</span> allele_freqs,  <span class="dt">k =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>), <span class="dt">collapse =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb8-3" title="3">  <span class="dt">FS =</span> <span class="kw"><a href="../reference/dbExpect.html">dbExpect</a></span>(<span class="dt">probs =</span> allele_freqs,  <span class="dt">k =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">1</span>)<span class="op">/</span><span class="dv">4</span>, <span class="dt">collapse =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb8-4" title="4">  <span class="dt">FC =</span> <span class="kw"><a href="../reference/dbExpect.html">dbExpect</a></span>(<span class="dt">probs =</span> allele_freqs,  <span class="dt">k =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">3</span>)<span class="op">/</span><span class="dv">4</span>, <span class="dt">collapse =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb8-5" title="5">  <span class="dt">PC =</span> <span class="kw"><a href="../reference/dbExpect.html">dbExpect</a></span>(<span class="dt">probs =</span> allele_freqs,  <span class="dt">k =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>), <span class="dt">collapse =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb8-6" title="6">  <span class="dt">AV =</span> <span class="kw"><a href="../reference/dbExpect.html">dbExpect</a></span>(<span class="dt">probs =</span> allele_freqs,  <span class="dt">k =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span>)<span class="op">/</span><span class="dv">2</span>, <span class="dt">collapse =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb8-7" title="7">)</a></code></pre></div>
<p><img src="db_vignette_files/figure-html/unnamed-chunk-10-1.png" width="768"></p>
<p>The function <code>optim.relatedness</code> can be used to estimate the fraction of comparisons between first-cousins, avuncular, parent-child and full-subling relatives in the database.</p>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-brenner2007">
<p>Brenner, C. 2007. “Arizona DNA Database Matches.” <a href="http://dna-view.com/ArizonaMatch.htm">http://dna-view.com/ArizonaMatch.htm</a>.</p>
</div>
<div id="ref-curran2007">
<p>Curran, JM, SJ Walsh, and J Buckleton. 2007. “Empirical Testing of Estimated DNA Frequencies.” <em>Forensic Science International: Genetics, 1(3-4), 267–272.</em></p>
</div>
<div id="ref-kaye2009">
<p>Kaye, DH. 2009. “Trawling DNA Databases for Partial Matches: What Is the FBI Afraid of?” <em>Cornell Journal of Law and Public Policy, 19(1)</em>.</p>
</div>
<div id="ref-mueller2008">
<p>Mueller, LD. 2008. “Can Simple Population Genetic Models Reconcile Partial Match Frequencies Observed in Large Forensic Databases?” <em>Journal of Genetics, 87(2), 101–108.</em></p>
</div>
<div id="ref-troyer2001">
<p>Troyer, K, T Gilboy, and B Koeneman. 2001. “A Nine STR Locus Match Between Two Apparently Unrelated Individuals Using AmFlSTR Profiler Plus and Cofiler.” <em>In Genetic Identity Conference Proceedings, 12th International Symposium on Human Identification.</em></p>
</div>
<div id="ref-tvedebrink2010">
<p>Tvedebrink, T. 2010. “Statistical Aspects of Forensic Genetics – Models for Qualitative and Quantitative STR Data.” <em>Ph.D. thesis, Department of Mathematical Sciences, Aalborg University.</em></p>
</div>
<div id="ref-tvedebrink2012">
<p>Tvedebrink, T, PS Eriksen, JM Curran, HS Mogensen, and N Morling. 2012. “Analysis of Matches and Partial-Matches in a Danish DNA Reference Profile Data Set.” <em>Forensic Science International: Genetics 6(3), 387-392.</em></p>
</div>
<div id="ref-weir2004">
<p>Weir, BS. 2004. “Matching and Partially-Matching DNA Profiles.” <em>Journal of Forensic Sciences, 49(5), 1009–1014.</em></p>
</div>
<div id="ref-weir2007">
<p>———. 2007. “The Rarity of DNA Profiles.” <em>The Annals of Applied Statistics, 1(2), 358–370.</em></p>
</div>
<div id="ref-wiki_birthday">
<p>2010. “Birthday Problem.” <a href="http://en.wikipedia.org/wiki/Birthday_problem">http://en.wikipedia.org/wiki/Birthday_problem</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#dna-evidence-matches-and-partial-matches">DNA evidence, matches and partial matches</a></li>
      <li><a href="#dna-database-comparison-exercises">DNA database comparison exercises</a></li>
      <li><a href="#the-birthday-problem">The birthday problem</a></li>
      <li><a href="#using-the-package-dnatools">Using the package <code>DNAtools</code></a></li>
      </ul>
</li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Torben Tvedebrink, James Curran, Mikkel Meyer Andersen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
